org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/AnalyzeProjectJob.java

```diff
@@ -115,7 +155,64 @@ protected IStatus run(SonarRunnerFacade runner, final IProgressMonitor monitor)
     return Status.OK_STATUS;
   }
 
-  private void analysisCompleted(Properties properties, final IProgressMonitor monitor) {
+  private void updateMarkers(Map<IResource, List<Issue>> issuesPerResource) throws CoreException {
+    ITextFileBufferManager iTextFileBufferManager = FileBuffers.getTextFileBufferManager();
+    if (iTextFileBufferManager == null) {
+      return;
+    }
+    PreviousMarkerCache markerCache = new PreviousMarkerCache(this.request);
+    for (Entry<IResource, List<Issue>> resourceEntry : issuesPerResource.entrySet()) {
+      IResource r = resourceEntry.getKey();
+      try {
+        List<IMarker> previousMarkers = markerCache.getPrevious(r);
+        List<Issue> rawIssues = resourceEntry.getValue();
+        if (r instanceof IFile) {
+          issueTrackingOnFile(iTextFileBufferManager, r, previousMarkers, rawIssues);
+        } else {
+          issueTracking(r, previousMarkers, rawIssues, null);
+        }
+      } catch (Exception e) {
+        SonarLintCorePlugin.getDefault().error("Unable to compute position of SonarLint marker on resource " + r.getName(), e);
+      }
+    }
+    markerCache.deleteUnmatched();
+  }
+
+  private void issueTrackingOnFile(ITextFileBufferManager iTextFileBufferManager, IResource r, List<IMarker> previousMarkers, List<Issue> rawIssues)
+    throws CoreException, BadLocationException {
+    IFile iFile = (IFile) r;
+    try {
+      iTextFileBufferManager.connect(iFile.getFullPath(), LocationKind.IFILE, new NullProgressMonitor());
+      ITextFileBuffer iTextFileBuffer = iTextFileBufferManager.getTextFileBuffer(iFile.getFullPath(), LocationKind.IFILE);
+      IDocument iDoc = iTextFileBuffer.getDocument();
+
+      issueTracking(r, previousMarkers, rawIssues, iDoc);
+
+    } finally {
+      try {
+        iTextFileBufferManager.disconnect(iFile.getFullPath(), LocationKind.IFILE, new NullProgressMonitor());
+      } catch (CoreException e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1166)